MK3 3 9 0 missing translations

3d-gussner opened the pull request:
## Translations for new messages for the FW 3.9.0 final.

### Done
- I added all "not translated" messages for MK2.5/MK2.5S/MK3/MK3S builds.
- Translated
   - [x] German
   - [x] French
   - [x] Spanish

I am not a French/Spanish native speaker, so please feel free to comment if translations aren't correct or can be improved.
Keep in mind that LCD messages often are limited in characters and rows. (c=xx r=yy)

### ToDo
- [x] Czech
- [x] Italian
- [x] Polish
- As soon we have all translations I can start to test these.
- Update PR to be ready to be pulled.

**This PR is based by accident on https://github.com/prusa3d/Prusa-Firmware/pull/2638 . Sorry for that.**

PFW-1112

3d-gussner committed:
Merge remote-tracking branch 'origin/patch-4' into MK3_3_9_0_translations

3d-gussner committed:
Add and update missing translations
- updated in Firmware/ files the missing `c=xx` column and `r=yy` rows.
- added missing translations to lang/lang_en*.txt

Everyone is developing and adding messages to serial and especially to LCD  PLEASE add `//// c=xx` or `//// c=xx r=yy` comments.
Preparing translations files without that information is a pain in the ... and takes way more time for somebody else
to review to code as it would take you.

3d-gussner committed:
No need to have `MSG_abcde` again in comments `////` in `messages.c`

3d-gussner committed:
German translation

3d-gussner committed:
Missed a space

3d-gussner committed:
Use the same format as somewhere else

3d-gussner committed:
French translation.
I am not a native French speaking person, so please excuse my mistakes I may have done.

3d-gussner committed:
Spanish translation.
I am not a native Spanish speaking person, so please excuse my mistakes I may have done.

DRracer committed:
CZ translation

DRracer commented on lang/lang_en_de.txt:
```diff
@@ -2,6 +2,14 @@
+
+#MSG_04_OR_NEWER c=18
+" 0.4 or newer"
+" 04. oder neuer"
```
typo 04. instead of 0.4

DRracer commented on lang/lang_en_de.txt:
```diff
@@ -568,7 +636,7 @@
 #MSG_AUTO_POWER
 "Auto power"
-"\x00"
+"Auto leist"
```
shouldn't it be Auto **L**eist?

DRracer commented on lang/lang_en_es.txt:
```diff
@@ -94,10 +106,22 @@
 
+#MSG_BELTTEST c=17
+"Belt test        "
+"Test cinturon"
```
for some reason there are additional spaces in the EN text

3d-gussner replied:
I don't know why there are sapces but i will update these

leptun replied:
The spaces are used so that the whole 20 characters are filled. Otherwise, the missing characters will use what was before on the display.
Other status screen messages have better implementations that don’t need this hack. Might be worth looking into it to save some more bytes of progmem

3d-gussner replied:
For the moment I keep the spaces
Add to list 3.9.1

DRracer commented on lang/lang_en_es.txt:
```diff
@@ -306,6 +350,14 @@
+
+#c=18
+"FS v0.4 or newer"
+"FS v0.4 o +nueva"
```
typo "+"?

3d-gussner replied:
+ = mas ;-)

3d-gussner replied:
Also here we could reuse existing messages to save some space

DRracer commented on lang/lang_en_fr.txt:
```diff
@@ -2,6 +2,14 @@
+
+#MSG_04_OR_NEWER c=18
+" 0.4 or newer"
+"0.4 ou plus recent"
```
missing space - ok, the text needs 18 chars. The space may be added if we find a shorter translation

3d-gussner replied:
I would prefer NOT to have a starting space as other Support messages.
Only one that are
Firmware:
Fil. sensor v.:

We could reuse "Fil. sensor" message here ... not sure that the `v.` is really needed.

3d-gussner committed:
Fix typos

3d-gussner committed:
Merge branch 'MK3_3_9_0_translations' of https://github.com/3d-gussner/Prusa-Firmware into MK3_3_9_0_translations

3d-gussner committed:
Another fix
It is
Dimmwert and not Dim Wert

3d-gussner committed:
Fix issues reported by `lang-check.py`

3d-gussner committed:
Add "difficult" messages containing `%`

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -892,7 +892,7 @@
 #c=20 r=4
 "Please unload the filament first, then repeat this action."
-"\x00"
+"Veuillez d'abord decharger le filament, puis repetez cette action."
```
I believe that this translation could be shortened to `SVP, dechargez le filament, puis faire cela encore`
Edit: missed a "z" in "decharge***z***"

3d-gussner replied:
@awenelo Have you seen any other french translation that can be shorten/optimized?

3d-gussner replied:
@awenelo Can you review other French translations as well
```
#MSG_WIZARD_REPEAT_V2_CAL c=20 r=7
"Do you want to repeat last step to readjust distance between nozzle and heatbed?"
"Voulez-vous repeter la derniere etape pour reajuster la distance entre la buse et le plateau chauffant?"
```
We will discuss these changes with the translators.

carlin57 replied:
> I believe that this translation could be shortened to SVP, dechargez le filament, puis faire cela encore
> Edit: missed a "z" in "dechargez"

SVP, dechargez le filament et recommencer

carlin57 replied:
or "SVP, dechargez le filament et reessayez

carlin57 replied:
> @awenelo Can you review other French translations as well
> #MSG_WIZARD_REPEAT_V2_CAL c=20 r=7
> "Do you want to repeat last step to readjust distance between nozzle and heatbed?"
> "Voulez-vous repeter la derniere etape pour reajuster la distance entre la buse et le plateau chauffant?"
> 
> We will discuss these changes with the translators.

"Voulez-vous renouveler l'etape pour reajuster la distance entre la buse et le plateau chauffant?"
"Voulez-vous renouveler l'etape pour reajuster la hauteur entre la buse et le plateau"

awenelo replied:
`renouveler` could also be translated to "renew", I think a better/shorter option would be `refaire`

carlin57 replied:
"Voulez-vous refaire l'etape pour reajuster la hauteur entre la buse et le plateau?

"refaire" it's ok for me

3d-gussner replied:
> SVP, dechargez le filament et reessayez

I take this as you both agreed on that.

3d-gussner replied:
It is `"Voulez-vous refaire l'etape pour reajuster la hauteur entre la buse et le plateau?"
` for #MSG_WIZARD_REPEAT_V2_CAL c=20 r=7

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -1187,6 +1294,10 @@
 
+#c=20 r=4
+"Verification failed, remove the filament and try again."
+"Echec de la verification, retirez le filament et reessayez."
```
This could be shortened to `Verification a echoue. Enlevez le filament et reessayez.`

3d-gussner replied:
Thanks again. I will wait a bit before changing it, maybe other will comment as well.

carlin57 replied:
"Verification en echec, retirez le filament et reessayez"

3d-gussner replied:
> Verification en echec, retirez le filament et reessayez

Take this one.

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -42,6 +50,10 @@
+
 #MSG_PRESS c=20
 "and press the knob"
 "et pressez le bouton"
```
This translation translates to "and squeeze the button." I believe that a better translation would be `et poussez le bouton`

3d-gussner replied:
@awenelo found other "press the knob" and it has been traslated to `et appuyez sur le bouton` Is that okay?

awenelo replied:
Yes, `appuyez sur` would work for "press".
A few other translations that I noticed:

`lit` translates to "bed"(as in the bed you sleep on) or is the 3rd person present tense of the verb "to read" (he/she reads)

- If it is supposed to be "heated bed", `plateau chauffant`(heating plate) would be a better option
- If it is supposed to be "Please read", `lis`(2nd person imperative) would be a better conjugation.

I see three different translations of "Steel sheets":
- `Feuilles d'acier`, which is "Steel Sheets" (as in paper sheets) or "Steel leaves" (as in tree leaves)
- `Plaque en acier`, which is "Steel Plates"
- `Plaque d'impression`, which is "Printing Plates"

`Ventilo tete` translates to "head fan" (I read this to be more a fan for your head). As there are no other fans that I am aware of, we can probably change it to `Ventilo`(familiar version of fan) or `ventilateur`(more formal version of fan)

When reading `Mesure de la hauteur de reference du point de calibration`, I take it as a command to Measure the reference height of the calibration point. Another way that this could be understood is that it is displaying, somewhere with that text, the reference height of the calibration point. Also,  in french, `calibration` is the age of a historical object. A better translation would probably be `Je mesure la hauteur de reference du point de calibrage `
edit: Oops, `du`, not `de la`

Also, `Calibration`, when a translation of the verb "to calibrate" should be some conjugation of the verb `Calibrer`. If its a command ("calibrate") it should be translated to `calibre`, and if it's "I'm calibrating", `Je calibre`.

carlin57 replied:
When reading Mesure de la hauteur de reference du point de calibration, I take it as a command to Measure the reference height of the calibration point. Another way that this could be understood is that it is displaying, somewhere with that text, the reference height of the calibration point. Also, in french, calibration is the age of a historical object. A better translation would probably be Je mesure la hauteur de reference de la point de calibrage 

"je mesure la hauteur de reference du point de calibrage"

3d-gussner replied:
> et appuyez sur le bouton

Used this.
I will try to implement the other comments you added here as well. Makes it bit more difficult as have to search for these and not commented at the translation.

3d-gussner replied:
Updated all `bed` to `plateau chauffant`

3d-gussner replied:
Updated `Steel Sheets` to `Plaques en acier`

3d-gussner replied:
```
#MSG_SELFTEST_EXTRUDER_FAN c=20
"Left hotend fan?"
"Ventilo gauche?"
```
Hope that is clear

3d-gussner replied:
```
#MSG_MEASURE_BED_REFERENCE_HEIGHT_LINE1 c=60
"Measuring reference height of calibration point"
"Je mesure la hauteur de reference de la point de calibrage"
```

3d-gussner committed:
Updated MSG and German translation

3d-gussner committed:
removed a translation as it breaks the language selection

3d-gussner committed:
No need to wait until any-key is pressed

3d-gussner committed:
No need to wait any-key is pressed

3d-gussner committed:
Fixed two LF issues

3d-gussner committed:
Updated PO files
ready to be send to translators

DRracer commented on lang/lang-add.sh:
```diff
@@ -69,5 +69,5 @@ cat lang_add.txt | sed 's/^/"/;s/$/"/' | while read new_s; do
 done
 
-read x
+read read -t 5
```
why do you have duplicit 'read' here?

wavexx committed:
Add missing italian translations

wavexx committed:
Improve some existing italian translations

wavexx committed:
More italian fixes

wavexx committed:
More italian fixes

wavexx committed:
Add exceptions in editorconfig for po files to avoid recoding

3d-gussner committed:
Fix typo
Thanks @DRracer for pointing out

3d-gussner committed:
Merge branch 'MK3_3_9_0_translations' of https://github.com/3d-gussner/Prusa-Firmware into MK3_3_9_0_translations

3d-gussner committed:
Italian translation by @wavexx

3d-gussner committed:
Update po/new/*.po files

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into MK3_3_9_0_translations

3d-gussner committed:
Update after merging MK3 branch

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -94,10 +106,22 @@
 
+#MSG_BRIGHT c=6
+"Bright"
+"Brill."
```
"Bright" = "Clarte"

awenelo replied:
I believe that this "Bright" is not bright, but is a short form of brightness, so a better translation would be `luminosité` but, with the 6 character restriction, we can't use that. In this case, I believe the best option is `Brill.`

leptun replied:
It's for the backlight mode. You have these three options:
Mode Bright -- this is where MSG_BRIGHT is used
Mode Dim
Mode Auto

carlin57 replied:
> I believe that this "Bright" is not bright, but is a short form of brightness, so a better translation would be luminosité but, with the 6 character restriction, we can't use that. In this case, I believe the best option is Brill.

"clarté" and "luminosité" it's same  ( sorry for my English)
 
(les volets sont ouvert il y a plus de "luminosité" dans la pièce =les volets sont ouvert il y a plus de "clarté" dans la pièce)

carlin57 replied:
> It's for the backlight mode. You have these three options:
> Mode Bright -- this is where MSG_BRIGHT is used
> Mode Dim
> Mode Auto

it's for oled screen?

leptun replied:
@carlin57 No, it’s for MK3S LCD screens that have backlight control capability. Prusa started shipping those I think last summer.

carlin57 replied:
ok thanks,

carlin57 replied:
> It's for the backlight mode. You have these three options:
> Mode Bright -- this is where MSG_BRIGHT is used
> Mode Dim
> Mode Auto

Mode Bright = mode clair
Mode Dim = mode fonce (foncé)
Mode Auto= mode auto

carlin57 replied:
> I believe that this "Bright" is not bright, but is a short form of brightness, so a better translation would be luminosité but, with the 6 character restriction, we can't use that. In this case, I believe the best option is Brill.

ok sorry but in french "Brill." I don't understand ( yes my English it's very bad) but "Lumin." , "Clarte" , "Lumi." it's ok for me. now if you prefer "Brill." no probleme.

awenelo replied:
As this translation represents "Bright", not "Brightness", `Brilliant` is probably the best option(shortened to `Brill.`)

3d-gussner replied:
> As this translation represents "Bright", not "Brightness", `Brilliant` is probably the best option(shortened to `Brill.`)
Use this.

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -234,10 +262,26 @@
+#MSG_CUT_FILAMENT c=16
+"Cut filament"
+"Coupe filament"
+
```
cut filament =couper filament

carlin57 replied:
or "couper filament"

awenelo replied:
I believe that `coupure filament` translates to "a break in the filament" or "filament cut"(a cut made by filament). I believe that `Coupe` (imperative conjugation of "Couper"-to cut) `filament` is correct

carlin57 replied:
or "Filament coupe" ( filament coupé)

awenelo replied:
> cut filament =couper filament

I believe that `Couper` should be conjugated to `Coupez`(more polite) or `Coupe`(what we use for most translations).

3d-gussner replied:
Stays as I translated it.

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -476,6 +530,10 @@
+#c=20 r=6
+"Insert the filament (do not load it) into the extruder and then press the knob."
+"Veuillez inserer le filament (ne pour le charger) dans l'extrudeur, puis appuyez sur le bouton."
+
```
Veuillez inserer le filament ( ne le chargez pas) dans l'extrudeur, puis appuyez sur le bouton

3d-gussner replied:
> Veuillez inserer le filament ( ne le chargez pas) dans l'extrudeur, puis appuyez sur le bouton

Use this.

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -2,6 +2,14 @@
+
+#MSG_04_OR_NEWER c=18
+" 0.4 or newer"
+"0.4 ou plus recent"
```
" 0.3 or older"  = "0.3 ou +ancien"

"0.4 or newer" = "0.4 ou +recent"

3d-gussner replied:
> 0.4 ou +recent

Use this
and also updated the 0.3

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -94,10 +106,22 @@
 #MSG_BED_LEVELING_FAILED_POINT_LOW c=20 r=4
 "Bed leveling failed. Sensor didnt trigger. Debris on nozzle? Waiting for reset."
 "Echec bed leveling. Capt. non declenche. Debris sur buse? En attente d'un reset."
 
```
"Echec niveau lit, Capt. non declenche. Debris sur buse? En attente d'un reset

awenelo replied:
`Echec niveau lit` translates to "bed level failed"(as in you failed a video game level centred around a bed). A better translation would be `Capt. non declenche pendant je nivele le plateau.`

carlin57 replied:
"capt. non declenche pendant que je nivele le plateau."

awenelo replied:
Unless I miscounted, there is no space for a `que` there(4 row limit, 20 characters per row). I belive that the message is not greatly changed by missing it.

carlin57 replied:
"Capt. non declanche lorsque je nivele le plateau"

3d-gussner replied:
> "capt. non declenche pendant que je nivele le plateau."

Use this for the 1st sentence. 
Long messages c=20 r>1 aren't a big issue to make here r=5 or 6 even saving space is also needed. We try keep it short and clear.

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -166,7 +190,7 @@
 "Copy selected language?"
 "Copier la langue selectionne?"
 
```
"Copy selected language?"
"Copier la langue choisie?"

3d-gussner replied:
> Copier la langue choisie?

Use this.

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -280,6 +325,7 @@
+
 #
 "Filam. runouts"
 "Fins de filament"
```
"Fin de filament"

awenelo replied:
`Fin de filament` is "The end of the filament", `Fins de filament` is "Endings of filament". I believe that "Endings of filament" is a better representation of "Filament runouts.

carlin57 replied:
have 1 one end of filament or more?

awenelo replied:
This message appears under `Fail stats`, and is a total count of the number of times it has run out of filament during a print (detected an end of filament). So, this message should be plural, as, over the course of the printer's life, it will detect multiple ends of filament.

3d-gussner replied:
Not sure what to change here. as it is limited to c=14 (just updated after checking code)
It is now `Fins filament`

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -234,10 +262,26 @@
+
+#c=18
+"Cutting filament"
+"Coupe filament"
```
"Coupez filament"

awenelo replied:
`Coupez filament` and `Coupe filament` are commands to cut the filament. In this case, the printer is doing the cutting, so a better translation would be `Je coupe filament`.

carlin57 replied:
"Coupe du filament" ?

awenelo replied:
I believe that this is an informative message and is displayed when the MMU is cutting filament. I interpret `coupe du filament` as "cut some filament" or "cut of filament". Really all we're doing with this message is telling the user that the printer is doing work. I cannot think of an alternative way to tell the user that the printer is working while staying within the limits.

awenelo replied:
I believe the best option here is `Je coupe filament`.

carlin57 replied:
ok for me.

3d-gussner replied:
> Je coupe filament

Use this

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -261,6 +305,7 @@
 #MSG_FSENSOR_AUTOLOAD
 "F. autoload"
 "Autochargeur"
+
```
"F. autoload" = "F. autocharg."

3d-gussner replied:
> F. autocharg.

Use this.
Also update it to c=13 after looking into the source code

3d-gussner commented:
@awenelo and @carlin57 Wow what a nice job you both do, really appreciate your help.

As you are the experts, it would be great if you leave a 👍 for the translation you both agree on.

I will add your suggestions/translations tomorrow, if that is okay for you?

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -1311,6 +1422,10 @@
+# c=20
+"%s level expected"
+"\x00"
+
```
level expected = prevoir niveau ?

carlin57 commented on lang/lang_en_fr.txt:
```diff
@@ -1147,6 +1250,10 @@
 
+#MSG_TIMEOUT c=12
+"Timeout"
+"\x00"
```
timout= epuise ( épuisé, écoulé, la fin de...) ?

3d-gussner commented:
@awenelo 🦸‍♂️ and @carlin57 🦸‍♂️... Thanks again for the 50+ comments.

I will add your translations later to this PR. Thank you both sooo much  for the help.

We just have to wait for the polish translation and we are good to go, one step closer to FW3.9.0 final.🎉

3d-gussner commented:
As you are still commenting I will wait merging the changes.
Please leave a comment when you think it is ready.

3d-gussner committed:
Update French translation and some c=xx comments

Big thanks to @awenelo @carlin57 for helping with the french translations and their comments.

3d-gussner committed:
Update po files after French translation

3d-gussner committed:
Fixed most `lang-check.py` reported translation errors for Czech and German.
Two Czech have to be reviewed as these are too long.
One German is correct as it is shown in c=20 r=2 but is 1 char longer than this to split the message.
One German translation seams to be to long but have to review the actual max length

3d-gussner committed:
Fix `lang-check.py` Spanish translation errors
There have been quite lot TOO long messages,

Can't imagine that nobody every complained about that.

3d-gussner committed:
Fix `lang-check.py` Italian translations errors

3d-gussner committed:
Update not_tran and not_used files after fixing several translations

3d-gussner committed:
Some more error fixes and update of `po` files

Commit:
Polish translation

Commit:
Merge branch 'MK3_3_9_0_translations' of github.com:3d-gussner/Prusa-Firmware into MK3_3_9_0_translations

Commit:
Czech updated

3d-gussner committed:
Fix typo

3d-gussner committed:
Merge branch 'MK3_3_9_0_translations' of https://github.com/3d-gussner/Prusa-Firmware into MK3_3_9_0_translations

3d-gussner committed:
no need to translate `\x00` if it is the same

carlin57 reviewed (changes requested):
https://github.com/prusa3d/Prusa-Firmware/pull/2646#discussion_r423058152

"Je mesure la hauteur de reference du point de calibrage" ( not "de la")

Commit:
Polish: Runouts->Koniec

Commit:
Polish: Runouts->Konce f ... hopefully the last change

3d-gussner committed:
Added MK2.5/s auto power mode to eeprom doxygen

3d-gussner committed:
Final updates.

- Compiled all versions with multi-languages
- Compiled all versions with EN_ONLY
- updated all /lang/po/Firmware*.* files

3d-gussner commented:
@awenelo @carlin57 Could you please review my last changes in `lang_en_fr.txt`? I had to shorten some of these.
@wavexx can you review the `lang_en_it.txt`? I had to shorten some of these.

3d-gussner commented:
@awenelo @carlin57 the 🚀means that I changed it.
Also had to force some c=20 r=2 message to fit exactly as the row 3 and 4 have "Yes" / "No".

wavexx committed:
Add crlf attributes for po files

As done for editorconfig, this similarly forces git to handle
po files consistently in DOS format.

wavexx committed:
Further improvent of IT translations

wavexx commented:
Did some further tweaks. I fixed some more inconsistencies, but now that I noticed, the translation of "piastra d'acciaio" could be better translate as "foglio d'acciaio". It's not incorrect, but it's the equivalent to sheet, and more commonly used for this. The manual itself is a bit random on the exact term and could use "foglio" too.

wavexx commented:
I think I originally translated this:

https://github.com/3d-gussner/Prusa-Firmware/blob/5c31f82c49f64a3813870b38573f9ddda77de6b9/lang/po/Firmware_it.po#L471

and the next translation too. "FS Action" and "FS v0.4 or newer".
Did it get lost, or was it done on purpose?

3d-gussner committed:
Merge pull request #74 from wavexx/po_attributes

Add Po attributes

3d-gussner committed:
Merge pull request #75 from wavexx/it_translations_2

Further improvent of IT translations

3d-gussner committed:
Merge branch 'MK3_3_9_0_translations' of https://github.com/3d-gussner/Prusa-Firmware into MK3_3_9_0_translations

Updated French and Italian translations
Cleanup double useless `not_tran` and `not_used` txt files.

3d-gussner committed:
Updated translation
Added cleanup to PF-build.sh

Commit:
remove lang/not_tran* and lang/not_used mistakenly added into the PR

3d-gussner commented:
Even this PR is already merged I want to **THANK YOU** all for the great work you all did to make this possible:
🥇@awenelo for helping with the French translation.
🥇@carlin57 for helping with the French translation.
Really liked the interaction between you both and to agree on "best" translation. I learned a lot from these comments.
🥇@wavexx for providing the Italian translation, and updating some editor settings to keep the po files in an format for the translators using POEdit or similar applications.
🥇@DRracer for providing the Czech translation and getting the Polish as well.

I tried my best to update the Spanish translations.
Really hope that we get as awesome community members like you to also review Spanish in the future.

DRracer commented:
@awenelo @carlin57 please contact me via email, we'd like to reward you for your work on the translations.
